frontend/src/js/button/DownloadButton.tsx

```diff
@@ -13,8 +15,31 @@ const Link = styled("a")`
   line-height: 1;
 `;
 
+const fileTypeToIcon: Record<string, IconName> = {
+  ZIP: "file-archive",
+  XLSX: "file-excel",
+  PDF: "file-pdf",
+  CSV: "file-csv",
+};
+function getFileIcon(label: string): IconName {
+  // Editor Requests
+  if (label in fileTypeToIcon) {
+    return fileTypeToIcon[label];
+  }
+
+  // Forms
+  if (label.includes(".")) {
+    const ext = label.split(".").pop()?.toLocaleUpperCase();
+    if (!ext) return "file-download";
+    if (ext in fileTypeToIcon) {
+      return fileTypeToIcon[ext];
+    }
+  }
+  return "file-download";
+}
+
 interface Props extends Omit<IconButtonPropsT, "icon" | "onClick"> {
-  url: string;
+  url: ResultUrlsWithLabel;
```

Comment:
Let's please avoid `url.url` anywhere. Here you could do
```suggestion
  resultUrl: ResultUrlWithLabel;
```